test/model.updateOne.test.js

```diff
@@ -3051,6 +3051,23 @@ describe('model: updateOne: ', function() {
       await Person.updateOne({ name: 'Anakin' }, { name: 'The Chosen One' }).orFail();
     }, { message: 'No document found for query "{ name: \'Anakin\' }" on model "gh-11620"' });
   });
+  it('updateOne with top level key that starts with $ (gh-13786)', async function() {
+    const [major] = await start.mongodVersion();
+    if (major < 5) {
+      return this.skip();
```

Comment:
maybe consider annotating this with `Mongodb version 5 added support for top-level $ keys`